lib/api_auth/request_drivers/rest_client.rb

```diff
@@ -79,7 +81,9 @@ def authorization_header
     private
 
       def find_header(keys)
-        keys.map {|key| @headers[key] }.compact.first
+        keys.map do |key|
+          @headers[key] || @processed_headers[key]
+        end.compact.first
```

Comment:
And because the `@headers` variable contains the capitalized `processed_headers` this method can be the original version: https://github.com/mgomes/api_auth/blob/master/lib/api_auth/request_drivers/rest_client.rb#L83-L85